### PR TITLE
Rule manager y flows

### PR DESCRIPTION
**Pull request by timofei-durakov**

### CODE REVIEW

File: src-java/kilda-model/src/main/java/org/openkilda/model/cookie/FlowSegmentCookie.java [hunk 1]
```diff
@@ -34,11 +34,12 @@
     static final BitField FLOW_FORWARD_DIRECTION_FLAG = new BitField(0x4000_0000_0000_0000L);
     static final BitField FLOW_LOOP_FLAG              = new BitField(0x0008_0000_0000_0000L);
     static final BitField FLOW_MIRROR_FLAG            = new BitField(0x0004_0000_0000_0000L);
+    static final BitField Y_FLOW_FLAG                 = new BitField(0x0002_0000_0000_0000L);
```
Comment (niksv) [inline src-java/kilda-model/src/main/java/org/openkilda/model/cookie/FlowSegmentCookie.java:37]: Please Update `cookie.md`

File: src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/generator/flow/SingleTableIngressYRuleGenerator.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        FlowEndpoint ingressEndpoint = FlowSideAdapter.makeIngressAdapter(flow, flowPath).getEndpoint();
+        FlowSpeakerCommandData command = buildFlowIngressCommand(sw, ingressEndpoint);
+        if (command == null) {
+            return Collections.emptyList();
+        }
+        result.add(command);
+
+        SpeakerCommandData meterCommand = buildMeter(flowPath, config, sharedMeterId, sw);
+        if (meterCommand != null) {
+            addMeterToInstructions(flowPath.getMeterId(), sw, command.getInstructions());
```
Comment (niksv) [inline src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/generator/flow/SingleTableIngressYRuleGenerator.java:62]: I'm just wondering is it OK that we add this meter instruction for single table mode, but not for multi table?

**Review by niksv — APPROVED** [review]

File: src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/generator/flow/SingleTableIngressYRuleGenerator.java (hunk 2 above)
Comment (timofei-durakov) [reply src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/generator/flow/SingleTableIngressYRuleGenerator.java]: no, it's not ok, fixed

File: src-java/kilda-model/src/main/java/org/openkilda/model/cookie/FlowSegmentCookie.java (hunk 1 above)
Comment (timofei-durakov) [reply src-java/kilda-model/src/main/java/org/openkilda/model/cookie/FlowSegmentCookie.java:37]: added

File: src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/generator/flow/MultiTableIngressYRuleGenerator.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        }
+        return builder.build();
+    }
+
+    private Instructions buildInstructions(Switch sw, List<Action> actions) {
+        Instructions instructions = Instructions.builder()
+                .applyActions(actions)
+                .goToTable(OfTable.POST_INGRESS)
+                .build();
+        addMeterToInstructions(sharedMeterId, sw, instructions);
```
Comment (rozdy) [inline src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/generator/flow/MultiTableIngressYRuleGenerator.java:109]: Probably missing meter availability check before call `addMeterToInstructions`.

**Review by rozdy — APPROVED** [review]

File: src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/generator/flow/MultiTableIngressYRuleGenerator.java (hunk 3 above)
Comment (timofei-durakov) [reply src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/generator/flow/MultiTableIngressYRuleGenerator.java:109]: it's part of the method

**Pull request merged**